arangod/Aql/SkipResult.h

```diff
@@ -55,7 +55,11 @@ class SkipResult {
 
   auto didSkip(size_t skipped) -> void;
 
-  auto didSkipSubquery(size_t skipped, size_t depth) -> void;
+  // depthOffset is added to depth, except it won't underflow.
+  template<int depthOffset = 0>
+  requires(depthOffset == 0 ||
+           depthOffset == -1) auto didSkipSubquery(size_t skipped, size_t depth)
+      -> void;
```

Comment:
Formatting looks very strange. Could try this:
```suggestion
  template<int depthOffset = 0>
  requires(depthOffset == 0 || depthOffset == -1) \\
  auto didSkipSubquery(size_t skipped, size_t depth) -> void;
```
or alternatively you could try to move the requires clause after the signature.